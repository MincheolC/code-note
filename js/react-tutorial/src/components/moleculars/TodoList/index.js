import React from 'react';
import styled from 'styled-components';
import TodoItem from '../../atoms/TodoItem';
import { useTodoState } from '../../contexts/TodoContext';

const TodoListBlock = styled.div`
  flex: 1;
  padding: 20px 32px;
  padding-bottom: 48px;
  overflow-y: auto;
`;

function TodoList() {
  const state = useTodoState();

  return (
    <TodoListBlock>
      {state.todos.map(({ id, text, done }) => (
        <TodoItem key={id} id={id} text={text} done={done} />
      ))}
    </TodoListBlock>
  );
}

export default TodoList;
