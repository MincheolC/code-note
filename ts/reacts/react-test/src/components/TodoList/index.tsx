import React from 'react'
import TodoItem from '../TodoItem'
import { Todo } from '../../shared/types'

interface TodoListProps {
  todos: Array<Todo>,
  onRemove: (id: number) => void,
}

const TodoList: React.FC<TodoListProps> = ({ todos, onRemove }) => {
  return (
    <ul>
      {todos.map(todo => <TodoItem key={todo.id} todo={todo} onRemove={onRemove}/>)}
    </ul>
  )
}

export default React.memo(TodoList)