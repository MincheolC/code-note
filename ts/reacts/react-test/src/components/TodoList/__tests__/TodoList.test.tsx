import React from 'react';
import renderer, { ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';
import TodoList from '../';

test('Todo render properly', () => {
  const onRemove = () => {};
  const todos = [
    { id: 1, todo: 'test1'},
    { id: 2, todo: 'test2'}
  ];
  const component: ReactTestRenderer = renderer.create(
    <TodoList todos={todos} onRemove={onRemove}/>,
  );

  let tree = component.toJSON() as ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();
});