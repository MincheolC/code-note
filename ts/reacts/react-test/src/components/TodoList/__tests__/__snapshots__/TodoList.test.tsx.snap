// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo render properly 1`] = `
<ul>
  <li>
    <span>
      test1
    </span>
    <button
      onClick={[Function]}
    >
      x
    </button>
  </li>
  <li>
    <span>
      test2
    </span>
    <button
      onClick={[Function]}
    >
      x
    </button>
  </li>
</ul>
`;
